manticore/models.py

```diff
@@ -84,3 +84,32 @@ def strcmp(state, s1, s2):
                 ret = 0
 
     return ret
+
+
+def strlen(state, s):
+    """
+    strlen symbolic model.
+
+    Algorithm: Walks from end of string not including NULL building ITE tree when current byte is symbolic.
+
+    :param State state: current program state
+    :param int s: Address of string
+    :return: Symbolic strlen result
+    :rtype: Expression or int
+    """
+
+    cpu = state.cpu
+
+    if issymbolic(s):
+        raise ConcretizeArgument(1)
```

Comment:
i'm concerned with using ALL because if the constraints are sufficiently loose, this is a good way to hang mcore. also, it doesn't make very much sense for the string ptr to be symbolic here in the first place; i'd expect this nearly guarantees a segfault in real execution. since we default to minmax, we'll also probably MemoryException, which is the appropriate way to handle this. i'll leave it like this for now.